### PR TITLE
Sets up an examples collection

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @addyosmani @gauntface 

This uses Jekyll's collections functionality to generate a list of all the examples we have, and tweaks the `_examples/index.md` page to make it a bit more presentable.

It also re-adds Examples to the navigation bar. Let me know if it's not in the right place.

The `workbox-sw` example is a placeholder to see how the list looked when there were multiple entries, and it will be populated with a real example in a future PR.
screen shot 2017-05-12 at 11 51 30 pm

### CODE REVIEW

File: src/_examples/index.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <li><a href="{{ example.url }}">{{ example.module }}</a></li>
+  {% endif %}
+{% endfor %}
+</ul>
+
+# Standalone Projects
+
+These projects are using Workbox for their service worker:
+
+- iFixit PWA (`gulp`-based build process)
```
Comment (jeffposnick) [inline src/_examples/index.md]: FYI, https://ifixit-pwa.appspot.com/ and https://github.com/GoogleChrome/application-shell/tree/ifixit-pwa/ifixit-pwa have been updated to use the current `0.0.1` published libraries.

File: src/_examples/index.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+[[source](https://github.com/GoogleChrome/application-shell/tree/ifixit-pwa/ifixit-pwa)]
+[[demo](https://ifixit-pwa.appspot.com/)]
+- React HN (`npm scripts`-based build process)
+[[source](https://github.com/insin/react-hn/tree/sw-helpers)]
```
Comment (jeffposnick) [inline src/_examples/index.md]: @addyosmani, do you want me to file a PR against https://github.com/insin/react-hn/tree/sw-helpers to update to Workbox?

Comment (addyosmani) [reply src/_examples/index.md]: Go for it!

**Comment by addyosmani** [comment]
Content here LGTM. With respect to where this shows up in the nav, I think this is fine for launch. Post I would say Reference should have submenus for API docs, examples etc.

File: src/_examples/index.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+<ul>
+{% for example in site.examples %}
+  {% if example.module %}
+    <li><a href="{{ example.url }}">{{ example.module }}</a></li>
+  {% endif %}
+{% endfor %}
+</ul>
+
+# Standalone Projects
```
Comment (gauntface) [inline src/_examples/index.md]: Could we change this to: "Projects using Workbox"

File: src/_examples/index.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
+layout: page
+title: Examples
+---
+
+# Module-specific Examples
```
Comment (gauntface) [inline src/_examples/index.md]: "Examples of Workbox Modules"

File: src/_examples/index.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+</ul>
+
+# Standalone Projects
+
+These projects are using Workbox for their service worker:
+
+- iFixit PWA (`gulp`-based build process)
+[[source](https://github.com/GoogleChrome/application-shell/tree/ifixit-pwa/ifixit-pwa)]
+[[demo](https://ifixit-pwa.appspot.com/)]
+- React HN (`npm scripts`-based build process)
```
Comment (gauntface) [inline src/_examples/index.md]: Can we add images or something as well to these?

I'm imaging the list to look very dull.

Comment (addyosmani) [reply src/_examples/index.md]: Aye. DevTools device mode screenshots of the apps where we highlight Workbox entries in the Application panel (or just simple screenshots of the apps themselves) would make for nice visuals here.

**Comment by jeffposnick** [comment]
@addyosmani @gauntface 

I've been working on the **big** `workbox-sw` example tonight, and realized that this branch handn't be merged yet. I've pushed out another commit with more sample code and if I could trouble for you some feedback on what's under `_examples/workbox-sw` that would be much appreciated. The code there isn't final and not ready to be merged, but it would be good to see if you have any major complaints about the general approach before I go much further.

**Comment by jeffposnick** [comment]
(You all might want to hold off on spending too much time reviewing the latest commits yet. There's a lot of late-night-commit-quality code in there that I will be cleaning up.)

**Comment by addyosmani** [comment]
> (You all might want to hold off on spending too much time reviewing the latest commits yet. There's a lot of late-night-commit-quality code in there that I will be cleaning up.)

Started reading through the latest commits but will hold off until you feel its ready.

**Comment by jeffposnick** [comment]
I think I'm at a point where I've done most of what I can do in this PR, and may get more aggressive about merging as per @gauntface's push-it-out recommendations.

The sample SWs for both `workbox-sw` and `workbox-routing` are pretty extensively commented and touch on most of the functionality.

Here are some screenshots of how things are currently laid out:
screen shot 2017-05-14 at 9 33 47 pm
screen shot 2017-05-14 at 9 33 58 pm
screen shot 2017-05-14 at 9 41 08 pm

**Review by gauntface — APPROVED** [review]

**Pull request merged**